pages/add/form/detail-info-fields.js

```diff
@@ -1,6 +1,105 @@
+import React from 'react';
+import ReactTags from 'react-tag-autocomplete';
 import validator from './validator';
+import Service from '../../../js/service';
+
+let Tags = React.createClass({
+  getInitialState() {
+    return {
+      tags: [],
+      suggestions: []
+    };
+  },
+  componentDidMount: function() {
+    Service.tags
+      .get()
+      .then((data) => {
+        console.log(data);
+        let suggestions = data.map((tag,i) => {
+          return { id: i+1, name: tag };
+        });
+        this.setState({ suggestions });
+      })
+      .catch((reason) => {
+        console.error(reason);
+      });
+  },
+  handleDelete: function(i) {
+    const tags = this.state.tags.slice(0);
+    tags.splice(i, 1);
+    this.setState({ tags });
+  },
+  handleAddition: function(tag) {
+    const tags = [].concat(this.state.tags, tag);
+    this.setState({ tags });
+  },
+  getFilteredSuggestions: function() {
+    // show suggestions that haven't been selected yet
+    const { tags, suggestions } = this.state;
+    const tagNames = tags.map(tagObj => tagObj.name);
+
+    return suggestions.map((suggestion) => {
+      if (tagNames.indexOf(suggestion.name) > -1) return null;
+
+      return suggestion;
+    }).filter(suggestion => !!suggestion);
+  },
+  render: function() {
+
+    return <ReactTags
+              tags={this.state.tags}
+              suggestions={this.getFilteredSuggestions()}
+              allowNew={true}
+              handleDelete={this.handleDelete}
+              handleAddition={this.handleAddition}
+              handleDrag={this.handleDrag}
+              onChange={this.onChange}
```

Comment:
you need an `onChange` function still, which then "falls through" to `this.props.onChange` (which is handled by the form).
```
  onChange: function(evt) {
    // do a thing!
    this.setState({
      // set a thing!
    }, () => {
      let values = this.state.tags.slice();
      this.props.onChange(evt, values);
    });
  },
```